### PR TITLE
import: do not read the entire SST into memory

### PR DESCRIPTION
**Pull request by kennytm**
Signed-off-by: kennytm <kennytm@gmail.com>

## What have you changed? (mandatory)

Avoid reading the entire SST file into memory when preparing to uploading them to TiKV stores. Replaces the original code which uses `Vec<u8>` by a `Read` instance. 

This further reduces the chance of OOM killing tikv-importer in exchange for having to read the SST file twice. The rationale is the same as #4348.

## What are the type of the changes? (mandatory)

- Improvement (change which is an improvement to an existing feature)

## How has this PR been tested? (mandatory)

Tested manually using a data set with size 6.5 TB

## Does this PR affect documentation (docs) update? (mandatory)

No

## Does this PR affect tidb-ansible update? (mandatory)

No

## Refer to a related PR or issue link (optional)

Part 3/7 split off from #4245.

### CODE REVIEW

File: src/import/stream.rs [hunk 1]
```diff
@@ -109,31 +111,36 @@ impl<Client: ImportClient> SSTFileStream<Client> {
...
-            ssts.push(self.new_sst_file(info));
-        }
-
-        Ok(Some((range, ssts)))
+        Ok(Some((range, infos)))
     }
+}
 
-    fn new_sst_file(&self, info: SSTInfo) -> SSTFile {
+impl LazySSTInfo {
```
Comment (Connor1996) [inline src/import/stream.rs]: why not impl it right after the struct `LazySSTInfo`

File: src/import/import.rs [hunk 2]
```diff
@@ -224,10 +224,11 @@ impl<Client: ImportClient> SubImportJob<Client> {
             .name("import-sst-job".to_owned())
             .spawn(move || {
                 'OUTER_LOOP: while let Ok((range, ssts)) = rx.lock().unwrap().recv() {
-                    for sst in ssts {
+                    for lazy_sst in ssts {
+                        let mut sst = lazy_sst.into_sst_file().unwrap();
                         let id = counter.fetch_add(1, Ordering::SeqCst);
                         let tag = format!("[ImportSSTJob {}:{}:{}]", engine.uuid(), sub_id, id);
-                        let mut job = ImportSSTJob::new(tag, sst, Arc::clone(&client));
+                        let mut job = ImportSSTJob::new(tag, &mut sst, Arc::clone(&client));
```
Comment (Connor1996) [inline src/import/import.rs]: seems we can just move it

**Comment by kennytm** [comment]
PTAL @Connor1996

**Comment by Connor1996** [comment]
please fix the conflict

**Pull request merged**